include/envoy/network/listen_socket.h

```diff
@@ -31,7 +31,7 @@ class Socket {
    */
   virtual void close() PURE;
 
-  enum class SocketState { PreBind, PostBind };
+  enum class SocketState { PreBind, PostBind, Listening };
```

Comment:
I think at this point we need clear comments on each of these to explain when they happen in the socket lifetime and relative to each other. Not sure why we need a `Listening` in addition to `PostBind` for example.

Reply:
Agreed.

I'll put this in the comments, but the reason for the new state is that on macOS, you cannot set this option until after listen() has been called. PostBind is run between bind() and listen().

Can you think of a better name for PostBind, now that this 3rd state is added? Or just comment it heavily and keep the name?

Reply:
I think it's fine to just comment. As long as there is a clear order in the comments in the socket creation -> accept life cycle it's all good.

Reply:
SocketState is documented